cli/azd/pkg/output/ux/time.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package ux
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+// DurationAsText provides a slightly nicer string representation of a duration
+// when compared to default formatting in go, by spelling out the words hour,
+// minute and second and providing some spacing and eliding the fractional component
+// of the seconds part.
+func DurationAsText(d time.Duration) string {
```

Comment:
❤️ This is a really clean implementation that also delivers clean output. Really nice work here!

Reply:
As a fun fact, I actually started the implementation here by regexp matching on the output of `d.String()` which was actually a really nice way to implement the feature quickly and write the tests against it, but then I figured out a slightly less perlish way of capturing what I wanted.